From dfd89b118222410d881bef80f9690297fdf0bef2 Mon Sep 17 00:00:00 2001
From: Fakeman <55951275+fakemanoan@users.noreply.github.com>
Date: Sun, 23 Jul 2023 14:37:27 +0100
Subject: [PATCH 1/3] Add back NETD_DIRECT_CONNECT_RULE

We use old libsec-ril blobs which require this patch for mobile data to function properly

Change-Id: I6b08f5cc351477d81915d65ef1ac50cb5c72d02e
---
 build/soong/Android.bp     | 17 +++++++++++++++++
 config/BoardConfigSoong.mk |  2 ++
 config/common.mk           |  2 +-
 3 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index 01e9cc3a..201ab571 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -187,6 +187,23 @@ trust_usb_control {
     },
 }
 
+soong_config_module_type {
+    name: "needs_netd_direct_connect_rule",
+    module_type: "cc_defaults",
+    config_namespace: "lineageGlobalVars",
+    bool_variables: ["needs_netd_direct_connect_rule"],
+    properties: ["cppflags"],
+}
+
+needs_netd_direct_connect_rule {
+    name: "needs_netd_direct_connect_rule_defaults",
+    soong_config_variables: {
+        needs_netd_direct_connect_rule: {
+            cppflags: ["-DNEEDS_NETD_DIRECT_CONNECT_RULE"],
+        },
+    },
+}
+
 soong_config_module_type {
     name: "bootloader_message_offset",
     module_type: "cc_defaults",
diff --git a/config/BoardConfigSoong.mk b/config/BoardConfigSoong.mk
index 8061220b..8cb33762 100644
--- a/config/BoardConfigSoong.mk
+++ b/config/BoardConfigSoong.mk
@@ -32,6 +32,7 @@ SOONG_CONFIG_lineageGlobalVars += \
     additional_gralloc_10_usage_bits \
     bootloader_message_offset \
     gralloc_handle_has_reserved_size \
+    needs_netd_direct_connect_rule \
     target_health_charging_control_charging_path \
     target_health_charging_control_charging_enabled \
     target_health_charging_control_charging_disabled \
@@ -64,6 +65,7 @@ endif
 
 # Soong bool variables
 SOONG_CONFIG_lineageGlobalVars_gralloc_handle_has_reserved_size := $(TARGET_GRALLOC_HANDLE_HAS_RESERVED_SIZE)
+SOONG_CONFIG_lineageGlobalVars_needs_netd_direct_connect_rule := $(TARGET_NEEDS_NETD_DIRECT_CONNECT_RULE)
 SOONG_CONFIG_lineageGlobalVars_uses_egl_display_array := $(TARGET_USES_EGL_DISPLAY_ARRAY)
 SOONG_CONFIG_lineageNvidiaVars_uses_nvidia_enhancements := $(NV_ANDROID_FRAMEWORK_ENHANCEMENTS)
 SOONG_CONFIG_lineageQcomVars_supports_extended_compress_format := $(AUDIO_FEATURE_ENABLED_EXTENDED_COMPRESS_FORMAT)
-- 
2.25.1

